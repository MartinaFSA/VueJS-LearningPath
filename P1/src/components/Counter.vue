<template>
  <h2>Counter app made in section 4</h2>
  <section>
    <p>Click the "+1" button to add clicks</p>
    <p><span>{{counter}}</span> clicks</p>
    <div>
        <button @click="addOne" data-test-id="addOne">+1</button>
        <button @click="restartCounter" data-test-id="restart">Restart</button>
    </div>
  </section>
</template>

<script>
export default {
    name:'CounterApp',
    data() {
        return {
            counter: 0
        }
    },
    methods: {
        addOne() {
            this.counter++
        },
        restartCounter() {
            this.counter = 0
        }
    }
}
</script>

<style>
    span {
        color: green;
        font-size: 1.2rem;
    }
</style>