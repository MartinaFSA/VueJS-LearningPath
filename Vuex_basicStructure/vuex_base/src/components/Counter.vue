<template>
    <div>
        <h1>valor: {{ $store.state.counter.count }}</h1>
        <h1>valor: {{ countComputed }}</h1>
        <h2>valor: {{count}}</h2>
        <p>{{lastMutation}}</p>
        <button @click="increment">+5</button>
    </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
    name: 'Counter-app',
    computed: {
        countComputed(){
            return this.$store.state.counter.count;
        },
        ...mapState('counter', ['count', 'lastMutation'])
    },
    methods:{
        increment() {
            this.$store.commit('counter/increment');
        }
    }
}
</script>
