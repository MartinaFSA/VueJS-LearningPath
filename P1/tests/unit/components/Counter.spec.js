import { shallowMount } from '@vue/test-utils';
import Counter from '@/pages/Counter.vue';

describe('Counter component tests', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount( Counter );
    })
    
    test('Should match snapshot', () => {
        expect ( wrapper.html() ).toMatchSnapshot();
    });

    test('When "+1" is clicked, counter must go up one number', async() => {

        const btn_addOne = wrapper.find('[data-test-id="addOne"]')
        await btn_addOne.trigger('click');
        
        const counter = wrapper.find('span').text();

        expect ( counter ).toBe( '1' );
    });

    test('Should restart the counter to 0', async() => {

        const btn_addOne = wrapper.find('[data-test-id="addOne"]')
        await btn_addOne.trigger('click');
        await btn_addOne.trigger('click');

        const btn_restart = wrapper.find('[data-test-id="restart"]')
        await btn_restart.trigger('click');
        
        const counter = wrapper.find('span').text();

        expect (counter).toBe( '0' );
    });
})
