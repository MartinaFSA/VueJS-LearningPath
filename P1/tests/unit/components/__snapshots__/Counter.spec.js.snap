// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component tests Should match snapshot 1`] = `
<h2>Counter app made in section 4</h2>
<section>
  <p>Click the "+1" button to add clicks</p>
  <p><span>0</span> clicks</p>
  <div><button data-test-id="addOne">+1</button><button data-test-id="restart">Restart</button></div>
</section>
`;
